### PR TITLE
feat: HTTP transport add prefix

### PR DESCRIPTION
**Pull request by eloycoto**
Currently yggd supports MQTT prefix, in this way adds a prefix to the
MQTT topic. This commit adds the same feature for HTTP, so adds the
prefix to the http endpoint.

Signed-off-by: Eloy Coto <eloy.coto@acalustra.com>

### CODE REVIEW

File: internal/transport/http.go [hunk 1]
```diff
@@ -98,5 +100,6 @@ func (t *HTTP) send(message []byte, channel string) error {
 }
 
 func (t *HTTP) getUrl(direction string, channel string) string {
-	return fmt.Sprintf("http://%s/%s/%s/%s", t.server, channel, t.clientID, direction)
+	path := filepath.Join(yggdrasil.TopicPrefix, channel, t.clientID, direction)
```
Comment (jakub-dzon) [inline internal/transport/http.go:103]: How will that be used by Flotta?

Comment (eloycoto) [reply internal/transport/http.go:103]:
```
sudo ./yggd  \
  --log-level info \
  --cert-file /tmp/client.pem \
  --key-file /tmp/client.key \
  --protocol http \
  --topic-prefix /api/k4e-management/v1 \
  --ca-root /home/eloy/dev/upstream/eloycoto/mtls_example/certs/rootCA.pem \
  --server localhost:8443
```

**Review by jakub-dzon — APPROVED** [review]
LGTM

**Review by subpop — CHANGES_REQUESTED** [review]
This looks great. Since we're already breaking the configuration file between 0.2 and 0.3, we should change this flag name to something more generic, so that it applies to both the MQTT and the HTTP transport.

* `--api-prefix`?
* `--path-prefix`?
* `--prefix`?

Any other suggestions?

**Comment by eloycoto** [comment]
+1 to path-prefix

**Review by strider — APPROVED** [review]
Looks good to me.

**Pull request merged**